controllers/hostingautoscalingpolicy/hostingautoscalingpolicy_controller.go

```diff
@@ -413,7 +428,14 @@ func (r *Reconciler) updateStatus(ctx reconcileRequestContext, hostingAutoscalin
 	return r.updateStatusWithAdditional(ctx, hostingAutoscalingPolicyStatus, "")
 }
 
-func (r *Reconciler) updateStatusAndReturnError(ctx reconcileRequestContext, status string, reconcileErr error) error {
+// Check the error code and update to error/reconciling status accordingly
+func (r *Reconciler) updateStatusAndReturnError(ctx reconcileRequestContext, reconcileErr error) error {
```

Comment:
Should be renamed to `updateAndReturnError` to match the other files.

Reply:
But it does update status

Reply:
Leaving as is